training/src/main/scala/com/airbnb/aerosolve/training/ScoreCalibrator.scala

```diff
@@ -92,8 +90,12 @@ object ScoreCalibrator {
                    learningRate : Double,
                    offset : Double,
                    slope : Double) : Array[Double] = {
-
-    val result = input
+    // First shuffle the data otherwise SGD won't work.
+    val data = input.map(x => (scala.util.Random.nextInt, x))
+      .sortByKey()
+      .map({case (r, x) => x})
+      .repartition(numBags)
```

Comment:
I don't believe this would perform the shuffling correctly because `repartition` uses hash partitioning by default IIRC. Hence it will give you the same deterministic repartition based on the has key of `(Double, Double)`. I think you can do `input.map(x => (math.rand, x)).sortBy(_._1, numPartitions  = numBags).values` instead.

Also is input cached by default? You may want to add that to avoid repeated I/O.